concrete/src/Entity/Command/Process.php

```diff
@@ -197,10 +197,12 @@ public function jsonSerialize()
             $hasDetails = $logger->logExists();
         }
 
-        $progress = 0;
-        if ($this->getBatch()) {
-            $remaining = $this->getBatch()->getTotalJobs() - $this->getBatch()->getPendingJobs();
-            $progress = round($remaining / $this->getBatch()->getTotalJobs(), 2) * 100;
+        $progress = 100;
+        $totalJobs = $this->getBatch()->getTotalJobs();
+        $pendingJobs = $this->getBatch()->getPendingJobs();
+        if ($this->getBatch() && $totalJobs > 0) {
```

Comment:
`if ($this->getBatch())` should be checked before calling `$this->getBatch()->getTotalJobs()` and `$this->getBatch()->getPendingJobs()`.

Reply:
👍 https://github.com/concrete5/concrete5/pull/10160